apps/VendingMachine/contracts/VendingMachine.ral

```diff
@@ -1,45 +1,41 @@
 import "std/nft_collection_interface"
 import "std/nft_collection_with_royalty_interface"
 
-Interface IRandomNumberGenerator{
-    @using(updateFields = true,checkExternalCaller = false)
-    pub fn getRandomNumber(userHash_: ByteVec) -> (U256)
-}
-
 Contract VendingMachine(
     foodsContractId: ByteVec,
     collectionOwner: Address,
-    collectionUri: ByteVec,
     mut nftBaseUri: ByteVec,
     mut totalSupply: U256, 
-    mut mintIsPaused: Bool // true
+    mut mintIsPaused: Bool, // true
+    mut mintedFoods: [U256;20]
 ) implements INFTCollectionWithRoyalty {
 
-    event NftMinted(minter: Address, id: ByteVec, index: U256)
+    event NftMinted(minter: Address, contractIds: ByteVec, mintAmount: U256,startingIndex: U256)
 
-    // const CollectionUri = b`Vending Machine`
+    const CollectionUri = b`Vending Machine`
```

Comment:
this should be a url to a collection json file i believe

Reply:
Good catch, I will make it a mutable. I thought it was a collection name